src/pointwise_loglikelihoods.jl

```diff
@@ -0,0 +1,40 @@
+function _pd_diag_inv(A::PDMats.AbstractPDMat)
+    T = typeof(float(oneunit(eltype(A))))
+    I = Diagonal(ones(T, axes(A, 1)))
+    return PDMats.invquad(A, I)
+end
+
+function pointwise_loglikelihood!(
+    logl::AbstractVector{<:Real}, y::AbstractVector{<:Real}, dist::Distributions.MvNormal
+)
+    Σ = cov(dist)
+    λ = _pd_diag_inv(Σ)
+    g = Σ \ (y - mean(dist))
+    return @. logl = (log(λ) - g^2 / λ - log2π) / 2
+end
+
+function pointwise_loglikelihood!(
+    logl::AbstractVector{<:Real},
+    y::AbstractVector{<:Real},
+    dist::Distributions.MvNormalCanon,
+)
+    J = invcov(dist)
+    λ = diag(J)
+    cov_inv_y = J * y
+    return @. logl = (log(λ) - (cov_inv_y - dist.h)^2 / λ - log2π) / 2
+end
+
+function pointwise_loglikelihoods(
+    obs::AbstractArray{<:Real,N},
```

Comment:
For consistency with our `loo_pit` signature:
```suggestion
    y::AbstractArray{<:Real,N},
```